datafusion/core/tests/fuzz_cases/equivalence/ordering.rs

```diff
@@ -158,3 +159,62 @@ fn test_ordering_satisfy_with_equivalence_complex_random() -> Result<()> {
 
     Ok(())
 }
+
+#[test]
+fn test_ordering_satisfy_on_data() -> Result<()> {
```

Comment:
Maybe it would help to add some comments about the rationale for this test -- specifically I think it is showing that data sorted on `[a,b,c,d]` or `[a,c,b,d]` is not also sorted on `[a ASC, b ASC, d ASC]`

Reply:
I have added a comment to explain rationale. Also, I put a link to original discussion for background. Thanks @alamb